cirq-core/cirq/devices/thermal_noise_model.py

```diff
@@ -228,10 +230,38 @@ def __init__(
         _validate_rates(qubits, rate_dict)
         self.gate_durations_ns: dict[type, float] = gate_durations_ns
         self.rate_matrix_GHz: dict[cirq.Qid, np.ndarray] = rate_dict
+        self.heat_rate_GHz = heat_rate_GHz
+        self.cool_rate_GHz = cool_rate_GHz
+        self.dephase_rate_GHz = dephase_rate_GHz
         self.require_physical_tag: bool = require_physical_tag
         self.skip_measurements: bool = skip_measurements
         self._prepend = prepend
 
+    def _value_equality_values_(self):
+        gate_times = tuple(sorted((k, v) for k, v in self.gate_durations_ns.items()))
+        rates = tuple(
+            sorted(
+                ((q, tuple(m.reshape(-1))) for q, m in self.rate_matrix_GHz.items()),
+                key=lambda x: repr(x[0]),
```

Comment:
The `self.rate_matrix_GHz` values are derived from other `__init__` arguments, which are also stored as object attributes.  Please use only the `__init__` arguments and `qubits = set(self.rate_matrix_GHz.keys())` for the result.

Reply:
Ack.  Here we compute values for equality.  Different `__init__` arguments (e.g., `heat_rate_GHz=None` vs `heat_rate_GHz=0.0`) can produce equal ThermalNoiseModel instances, so it is better to use `self.rate_matrix_GHz` here.